pkg/appgateserver/server.go

```diff
@@ -173,6 +173,9 @@ func (app *appGateServer) ServeHTTP(writer http.ResponseWriter, request *http.Re
 	path := request.URL.Path
 	serviceId := strings.Split(path, "/")[1]
 
+	// Remove the serviceId from the request path which is specific AppGateServer.
```

Comment:
1. #PUC why w/ an example
2. Consider creating a helper for this. There's a lot of implicit knowledge in this.